server/src/scheduler.rs

```
@@ -139,27 +139,35 @@ impl Scheduler {
                 }
                 let task =
                     task.ok_or(anyhow!("task not found: {}", task_finished_event.task_id))?;
-                let compute_graph = self
+
+                let compute_graph_version = self
                     .indexify_state
                     .reader()
-                    .get_compute_graph(&task.namespace, &task.compute_graph_name)
+                    .get_compute_graph_version(
+                        &task.namespace,
+                        &task.compute_graph_name,
+                        task.graph_version,
+                    )
                     .map_err(|e| {
-                        error!("error getting compute graph: {:?}", e);
+                        error!("error getting compute graph version: {:?}", e);
                         e
                     })?;
-                if compute_graph.is_none() {
+                if compute_graph_version.is_none() {
                     error!(
-                        "compute graph not found: {:?} {:?}",
+                        "compute graph version not found: {:?} {:?}",
                         task.namespace, task.compute_graph_name
                     );
                     return Ok(None);
                 }
-                let compute_graph = compute_graph.ok_or(anyhow!(
-                    "compute graph not found: {:?} {:?}",
+                let compute_graph_version = compute_graph_version.ok_or(anyhow!(
```

could this be combined with line 155 as a match statement?

---

like match None => {} and compute_graph_version => {}

---

this might be a stylistic comment - I see the definition for `get_compute_graph_version` has Result<Option<..>>`

---

This is related to the previous comment and to code style.

1) We actually want to return Ok(None) in this case since if the compute graph gets deleted, it is not an error if it is not found.
2) Since at this line compute_graph_version is an Option, we need a way to get the `Some` of the option. We could use a match statement here, but doing so would make all of our code more nested.

Line 152 is an actual error we got retrieving the graph version.
Line 155 is the graph version is not found we don't want to return an error
Line 162 is we just checked that the graph version cannot be None, in order to not use a match statement and have nested code, this time we return an error if it is None. (We could unwrap as well since we just checked and we know it cannot be None, but a panic in the scheduler is bad - and we try to not use unwrap).